src/include/optimizer/physical_operators.h

```diff
@@ -1155,5 +1159,761 @@ class Distinct : public OperatorNode<Distinct> {
+class DropTable : public OperatorNode<DropTable> {
+ public:
+  /**
+   * @param table_oid OID of the table to be dropped
+   * @return
+   */
+  static Operator Make(catalog::table_oid_t table_oid);
+
+  bool operator==(const BaseOperatorNode &r) override;
+  common::hash_t Hash() const override;
+
+  /**
+   * @return the OID of the database we want to drop
+   */
+  const catalog::table_oid_t &GetTableOID() const { return table_oid_; }
+
+ private:
+  /**
+   * OID of the table
+   */
+  catalog::table_oid_t table_oid_ = catalog::INVALID_TABLE_OID;
+};
+
+/**
+ * Physical operator for DropIndex
+ */
+class DropIndex : public OperatorNode<DropIndex> {
+ public:
+  /**
+   * @param index_oid OID of the index to be dropped
+   * @return
+   */
+  static Operator Make(catalog::index_oid_t index_oid);
+
+  bool operator==(const BaseOperatorNode &r) override;
+  common::hash_t Hash() const override;
+
+  /**
+   * @return the OID of the index we want to drop
+   */
+  const catalog::index_oid_t &GetIndexOID() const { return index_oid_; }
+
+ private:
+  /**
+   * OID of the table
+   */
+  catalog::index_oid_t index_oid_ = catalog::INVALID_INDEX_OID;
+};
+
+/**
+ * Physical operator for DropSchema
+ */
+class DropSchema : public OperatorNode<DropSchema> {
+ public:
+  /**
+   * @return
+   */
+  static Operator Make(catalog::namespace_oid_t namespace_oid);
+
+  bool operator==(const BaseOperatorNode &r) override;
+  common::hash_t Hash() const override;
+
+  /**
+   * @return the OID of the namespace we want to drop
+   */
+  const catalog::namespace_oid_t &GetNamespaceOID() const { return namespace_oid_; }
+
+ private:
+  /**
+   * OID of the table
+   */
+  catalog::namespace_oid_t namespace_oid_ = catalog::INVALID_NAMESPACE_OID;
+};
+
+/**
+ * Physical operator for DropTrigger
+ */
+class DropTrigger : public OperatorNode<DropTrigger> {
+ public:
+  /**
+   * @return
+   */
+  static Operator Make(catalog::db_oid_t database_oid, catalog::namespace_oid_t namespace_oid,
+                       catalog::trigger_oid_t trigger_oid, bool if_exists);
+
+  bool operator==(const BaseOperatorNode &r) override;
+  common::hash_t Hash() const override;
+
+  /**
+   * @return OID of the database
+   */
+  catalog::db_oid_t GetDatabaseOid() const { return database_oid_; }
+
+  /**
+   * @return OID of the namespace
+   */
+  catalog::namespace_oid_t GetNamespaceOid() const { return namespace_oid_; }
+
+  /**
+   * @return OID of the trigger to drop
+   */
+  catalog::trigger_oid_t GetTriggerOid() const { return trigger_oid_; }
+
+  /**
+   * @return true if "IF EXISTS" was used
+   */
+  bool IsIfExists() const { return if_exists_; }
+
+ private:
+  /**
+   * OID of the database
+   */
+  catalog::db_oid_t database_oid_;
+
+  /**
+   * OID of namespace
+   */
+  catalog::namespace_oid_t namespace_oid_;
+
+  /**
+   * OID of the trigger to drop
+   */
+  catalog::trigger_oid_t trigger_oid_;
+
+  /**
+   * Whether "IF EXISTS" was used
+   */
+  bool if_exists_;
```

Comment:
Do we need if_exists_ flags in the operators? We should be able to catch this at the Binder level in the future.

Reply:
> Do we need if_exists_ flags in the operators? We should be able to catch this at the Binder level in the future.

What is standard way for binder to handle this? When Binder detects that the trigger in the "drop trigger if exists" query does not exist, do we create a "no op" operator?

Reply:
Good question, I think we want some way to indicate back to the traffic cop that binding succeeded but execution of the query doesn't need to proceed any further. A noop operator is an interesting idea though for consistency purposes.